### ISSUE TITLE
Add push main permissions to releasers team

### ISSUE THREAD

**Issue by RafaelGSS** [nodejs/admin#825]
Recently, @ulisesGascon has been added and onboarded (v20.6.0) to the @nodejs/releasers team. This team has access to push to all vX.x-staging and vX.x branches, but does not have access to `main` - Ulises isn't a nodejs core member, so he couldn't finish the release.

I'm opening this issue to change the permissions to include push access to `main`.

**Comment by juanarbol** [comment]
Refs https://github.com/nodejs/node/commit/12354260db90f27a2c7f177bb161349718c53ccb

I had to push into main the release commit for him

**Comment by benjamingr** [comment]
The solution in the past was too make them a collaborator though relaxing that requirement would be good.

**Comment by RafaelGSS** [comment]
cc: @nodejs/releasers

**Comment by ljharb** [comment]
fwiw i was under the impression collab status was required to be a releaser; if it’s not, I’d love to help.

**Comment by benjamingr** [comment]
I like this solution better than what we did before (add to collaborators) since the two roles are distinct. We should also figure out how to better celebrate releasers to make sure that role is recognized.

So +1 on this change

**Comment by aduh95** [comment]
To be clear, we are discussing adding Write access on nodejs/node repo for @nodejs/releasers, correct? In addition to the ability of pushing release commits, this role comes with a bunch of vaguely related stuff: https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization#permissions-for-each-role

- approving a PR shows a green tick instead of grey, requesting changes shows a red cross, but those reviews would not "count" for ncu/the commit queue.
- Create, edit, apply/dismiss, delete labels
- Edit, hide, and delete anyone's comments on commits, pull requests, and issues
- Close, reopen, and assign all issues and pull requests
- it technically gives them the possibility to merge open PRs (but since we got rid of the green button on the UI, that's much less likely to happen by accident)

It would be important to clarify with releasers who are not collaborators what is acceptable and what is not, because IMO it's not obvious.
We have a precedent with the Moderation Team, its members are admin of the nodejs org, and are asked to not do anything that would not be allowed to them if they weren't member of the Moderation Team. Do we want to follow the same rule? Or on the contrary are we happy if they can help triaging issues?

I don't have an answer myself, just wondering what's the room temperature about that. I'm generally +1 on the proposal no matter what is decided for the details above.

**Comment by ljharb** [comment]
@RafaelGSS that link says the first step is an email to the TSC for approval. Should I file an issue requesting that?

**Comment by RafaelGSS** [comment]
Once you open the issue on nodejs/Release, I can handle the next steps!

**Comment by targos** [comment]
@aduh95 

> To be clear, we are discussing adding Write access on nodejs/node repo for https://github.com/orgs/nodejs/teams/releasers, correct?

No. The releasers team already has write access to the nodejs/node repo (it's required to push to release branches). This is only about adding permission to push to the main branch, which is currently protected by https://github.com/nodejs/node/settings/branch_protection_rules/12292636

**Comment by aduh95** [comment]
There are a few 👍  and explicit +1s from TSC members, no one has objected, unclear points have been clarified, and the issue has been opened for more than 7 days. I've proceeded with the suggested change, @nodejs/releasers are now allowed to push to the `main` branch of nodejs/node (for the purpose of pushing release commits).

**Comment by MylesBorins** [comment]
I want to come back and object to this decision and argue that it should be reverted. Releasers are the ultimate authority in what is Node.js, and I do not believe they can appropriate do their jobs as releasers if they are not collaborators. For LTS releases, the release team is the ultimate authority of what gets backported. How can someone have the trust to make those decisions but not the authority to approve PRs?

**Comment by aduh95** [comment]
@MylesBorins is my understanding correct that it's the LTS team that decides what goes into LTS releases, not the releasers team? We could decide being a collaborator is required to be on the LTS team.

**Comment by MylesBorins** [comment]
There is no distinction between the two anymore, anything implying otherwise is stale documentation

**Comment by BethGriggs** [comment]
Even with the distinction (which I agree we've not followed for some time, probably since https://github.com/nodejs/Release/pull/370), the typical flow of our releases where a commit _should_ land in Current first means that a releaser who is not in the LTS team can still have significant impact as to what ends up eligible for LTS.

> Generally, changes are expected to live in a Current release for at least 2 weeks before being backported. It is possible for a commit to land earlier at the discretion of the Release working group.

**Comment by richardlau** [comment]
> There is no distinction between the two anymore, anything implying otherwise is stale documentation

There is a distinction. **We do not require all members of the Release Working Group to be or become releasers.** Being a releaser is not a prerequisite for joining the working group nor for participating in discussions on release policy and decisions around what goes into LTS releases. This includes participating in Release WG votes -- I don't actually remember ever having a vote in the release WG (possibly because it's always been a small group).

Ultimately though the releaser has to sign the release which is a huge responsibility, so yes, the ultimate authority rests with the releaser in charge of the release.

TL;DR The LTS team is a subset of the Release WG whose interest is in LTS releases (policy and content). They are not the people  who do releases, which are the releasers team. Individuals can, of course, be on both teams.

A lot of the fuzziness goes way back -- in the beginning there was a Long Term Support (LTS) group (I forget if it was a chartered working group) that looked after LTS policy and what went into LTS releases. There was no group for current releases (in the beginning it wasn't felt necessary). Eventually the LTS group was expanded to also cover current releases and rechartered as the Release WG.

**Comment by UlisesGascon** [comment]
> There are a few 👍 and explicit +1s from TSC members, no one has objected, unclear points have been clarified, and the issue has been opened for more than 7 days. I've proceeded with the suggested change, https://github.com/orgs/nodejs/teams/releasers are now allowed to push to the main branch of nodejs/node (for the purpose of pushing release commits).

@aduh95 can you re-check the rule? I can't push to `main` the 20.7.0 release.

<img width="790" alt="Screenshot 2023-09-18 at 19 40 01" src="https://github.com/nodejs/admin/assets/5110813/5396da2c-c292-44dc-95e0-f9261bed965d">

**Comment by aduh95** [comment]
My sincere appologies, it looks like I forgot to hit the `Save changes` button. I've made the change, for good this time.
